src/Pages/SignupPage/Components/SignupForm.jsx

```diff
@@ -0,0 +1,242 @@
+import { useState } from "react";
+import { Link, useLocation, useNavigate } from "react-router-dom";
+import { useAuth, useModal, useAlert } from "../../../CustomHooks/";
+import axios from "axios";
+
+const Signupform = () => {
+  const intialState = {
+    firstName: "",
+    lastName: "",
+    email: "",
+    password: "",
+    confirmPassword: "",
+  };
+
+  const initialErrorState = {
+    error: false,
+    message: "",
+    emailError: false,
+    passwordError: false,
+    confirmPasswordError: false,
+  };
```

Comment:
You could have had this whole thing in a single state so there will be a single source of truth for the whole form and the form would be only changing and referencing only one state and one state change will dictate everything.like the following:

`const [signUpData, setSignUpData] = useState({
    input: {},
    error: "",
    hide: { pwd: true, confPwd: true },
  });`

inside this input property have what you have in the initial state state variable. for simplicity and readability